chains/evm/contracts/test/libraries/Internal.t.sol

```diff
@@ -0,0 +1,119 @@
+  }
+
+  function test_validateEVMAddress_reverts_onInvalidLength() public {
+    bytes memory invalidAddress = new bytes(31);
+    vm.expectRevert(abi.encodeWithSelector(Internal.InvalidEVMAddress.selector, invalidAddress));
+    s_helper.validateEVMAddress(invalidAddress);
+  }
+
+  function test_validateEVMAddress_reverts_onPrecompileAddress() public {
+    bytes memory precompileAddress = abi.encode(PRECOMPILE_ADDRESS);
+    vm.expectRevert(abi.encodeWithSelector(Internal.InvalidEVMAddress.selector, precompileAddress));
+    s_helper.validateEVMAddress(precompileAddress);
+  }
+
+  function test_validateEVMAddress_reverts_onOversizedAddress() public {
+    bytes memory invalidAddress = abi.encode(uint256(type(uint160).max) + 1);
+    vm.expectRevert(abi.encodeWithSelector(Internal.InvalidEVMAddress.selector, invalidAddress));
+    s_helper.validateEVMAddress(invalidAddress);
+  }
+
+  function test_validateEVMAddress_succeeds_onBoundaryAddresses() public {
+    bytes memory lowerBoundary = abi.encode(uint256(Internal.EVM_PRECOMPILE_SPACE));
+    s_helper.validateEVMAddress(lowerBoundary);
+
+    bytes memory upperBoundary = abi.encode(uint256(type(uint160).max));
+    s_helper.validateEVMAddress(upperBoundary);
+  }
+}
+
+contract Internal_validate32ByteAddress is InternalSetup {
+  function test_validate32ByteAddress_succeeds_onValidAddress() public {
+    s_helper.validate32ByteAddress(VALID_32_BYTE_ADDRESS, 0);
+  }
+
+  function test_validate32ByteAddress_reverts_onInvalidLength() public {
+    bytes memory invalidAddress = new bytes(31);
+    vm.expectRevert(abi.encodeWithSelector(Internal.Invalid32ByteAddress.selector, invalidAddress));
+    s_helper.validate32ByteAddress(invalidAddress, 0);
+  }
+
+  function test_validate32ByteAddress_reverts_onAddressBelowMinValue() public {
+    bytes memory belowMinAddress = abi.encode(uint256(500));
+    vm.expectRevert(abi.encodeWithSelector(Internal.Invalid32ByteAddress.selector, belowMinAddress));
+    s_helper.validate32ByteAddress(belowMinAddress, 1000);
+  }
+
+  function test_validate32ByteAddress_succeeds_onBoundaryMinValue() public {
+    uint256 minValue = 1000;
+    bytes memory exactMinAddress = abi.encode(minValue);
+    s_helper.validate32ByteAddress(exactMinAddress, minValue);
+  }
+
+  function test_validate32ByteAddress_reverts_onAptosPrecompileAddress() public {
+    bytes memory precompileAddress = abi.encode(Internal.APTOS_PRECOMPILE_SPACE - 1);
+    vm.expectRevert(abi.encodeWithSelector(Internal.Invalid32ByteAddress.selector, precompileAddress));
+    s_helper.validate32ByteAddress(precompileAddress, Internal.APTOS_PRECOMPILE_SPACE);
+  }
+}
+
+contract Internal_validateTVMAddress is InternalSetup {
+  function test_validateTVMAddress_succeeds_onValidAddress() public {
+    s_helper.validateTVMAddress(VALID_TVM_ADDRESS);
+  }
+
+  function test_validateTVMAddress_reverts_onShortLength() public {
+    bytes memory shortAddress = new bytes(35);
```

Comment:
It's not clear if this fails due to length checks or the zero address check if all bytes are 0